scipy/stats/_hypotests.py

```diff
@@ -1225,3 +1226,242 @@ def cramervonmises_2samp(x, y, method='auto'):
+           Method."
+           https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
+           28 November 2020.
+    .. [2] Abdi, Herve & Williams, Lynne. (2021). Tukey's Honestly Signiflcant
+           Difierence (HSD) Test.
+    .. [3] "One-Way ANOVA Using SAS PROC ANOVA & PROC GLM." SAS
+           Tutorials, 2007, www.stattutorials.com/SAS/TUTORIAL-PROC-GLM.htm.
+    .. [4] Kramer, Clyde Young. "Extension of Multiple Range Tests to Group
+           Means with Unequal Numbers of Replications." Biometrics, vol. 12,
+           no. 3, 1956, pp. 307-310. JSTOR, www.jstor.org/stable/3001469.
+           Accessed 25 May 2021.
+
+
+    Examples
+    --------
+    Here are some data comparing the time to relief of three brands of
+    headache medicine, reported in minutes. Data adapted from [3]_.
+
+    >>> from scipy.stats import tukeykramer
+    >>> group0 = [24.5, 23.5, 26.4, 27.1, 29.9]
+    >>> group1 = [28.4, 34.2, 29.5, 32.2, 30.1]
+    >>> group2 = [26.1, 28.3, 24.3, 26.2, 27.8]
+
+    We would like to see if the means between any of the groups are
+    significantly different. First, visually examine a box and whisker plot.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots(1, 1)
+    >>> ax.boxplot([group0, group1, group2])
+    >>> ax.set_xticklabels(["group0", "group1", "group2"])
+    >>> ax.set_ylabel("mean")
+    >>> plt.show()
+
+    From the box and whisker plot we can see overlap in the interquartile
+    ranges group 1 to group 2 and group 3, but we can apply the ``tukeykramer``
+    test to determine if the difference between means is significant.
+
+    >>> res = tukeykramer(group0, group1, group2)
+    >>> for ((i, j), p) in np.ndenumerate(res.pvalue):
+    >>>     # filter out self comparisons
+    >>>     if i != j:
+    >>>         print(f"({i} - {j}) {p:.03f}")
+    (0 - 1) 0.012
+    (0 - 2) 0.980
+    (1 - 0) 0.012
+    (1 - 2) 0.017
+    (2 - 0) 0.980
+    (2 - 1) 0.017
+
+    The null hypothesis is that each group has the same mean. We choose a
+    significance level of .05. The p-value for comparisons between ``group0``
+    and ``group1`` as well as ``group1`` and ``group2`` do not exceed .05, so
+    we reject the null hypothesis that they have the same means. The p-value of
+    the comparison between ``group0`` and ``group2`` exceeds .05, so we accept
+    the null hypothesis that there is not a significant difference between
+    their means.
+
+    There are also confidence intervals associated with the test statistic.
+    They are applicable for the 95% confidence level becuase the test was
+    applied with a .05 significance level. Print out the lower confidence
+    interval for this result:
+
+    >>> conf = res.confidence_interval()
+    >>> for ((i, j), l) in np.ndenumerate(conf.low):
+    >>>     # filter out self comparisons
+    >>>     if i != j:
+    >>>         h = conf.high[i,j]
+    >>>         print(f"({i} - {j}) {l:>6.3f} {h:>6.3f}")
```

Comment:
```suggestion
    ...     # filter out self comparisons
    ...     if i != j:
    ...         h = conf.high[i,j]
    ...         print(f"({i} - {j}) {l:>6.3f} {h:>6.3f}")
```

Reply:
Thanks. I wasn't sure why that was happening with the doctests.

Reply:
I've seen that error so many times in my own PRs : P Such a dumb rule, IMO.